### ISSUE TITLE
Not all new-style configuration settings for the skosmos:Configuration have an effect

### ISSUE THREAD

**Issue by rwalkerands** [NatLibFi/Skosmos#792]
`config.ttl.dist` has:

```
:config a skosmos:Configuration ;
    # ... several lines omitted ...
    # sparql-query extension, or "Generic" for plain SPARQL 1.1
    # set to "JenaText" instead if you use Fuseki with jena-text index
    skosmos:sparqlDialect "Generic" ;
```

But setting the default SPARQL dialect at this top level doesn't have any effect. To confirm: change this setting for the `skosmos:Configuration` to 
```
    skosmos:sparqlDialect "JenaText" ;
```
and observe that the queries _remain unchanged_.

The method ``Vocabulary::getSparql()`` seems to be correct:
```
$dialect = $this->resource->get('skosmos:sparqlDialect');
$dialect = $dialect ? $dialect->getValue() : $this->model->getConfig()->getDefaultSparqlDialect();
```

But it also seems that commit 6d28f04 did not update `model/GlobalConfig.php` completely to use the new configuration style; it still says:

```
public function getDefaultSparqlDialect()
{
    return $this->getLiteral('DEFAULT_SPARQL_DIALECT', 'Generic');
}
```

Perhaps it should be:
```
public function getDefaultSparqlDialect()
{
    return $this->getLiteral('skosmos:sparqlDialect', 'Generic');
}
```

And note also the surrounding methods in `GlobalConfig`: `getTemplateCache()`, `getFeedbackAddress()`, etc., which seem to have the same issue.

**Comment by kinow** [comment]
Oh, mea culpa @rwalkerands ! Will probably have some time this weekend to re-sync the repository and investigate it. But thanks already for always including plenty information in the bug reports. It makes fixing the code so much easier.

**Comment by osma** [comment]
Thanks a million for reporting and great analysis @rwalkerands ! This actually explains a couple of mysterious issues issues we've had with Skosmos 2.0 - we've been unable to deploy it to Finto.fi due to performance issues but didn't yet have time to investigate in depth. Getting the SPARQL dialect wrong is probably a major factor in that.

**Comment by kinow** [comment]
Oh, my reply didn't go to GitHub, sorry. Here's my message that goes before @osma's comment above:

>Super sorry for the regressions pre and post the 2.0 release Osma.
>Going to find some time to work on the issues reported, and try to increase the test coverage :(
>Cheers
>Bruno

And finished the fix locally. To try to prevent further regressions, added more tests, and now GlobalConfig has 100% coverage. Re-wrote the `GlobalConfigTest` which previously had some settings being tested with default values, and others with values from `testconfig.ttl`.

Now, `testconfig.ttl` is the opposite of `config.ttl.dist`. And `GlobalConfigTest` starts by testing all the methods of `GlobalConfig` to make sure it's getting the values from `testconfig.ttl`.

Further ahead in `GlobalConfigTest`, it is covering the possible exceptions (inexistent file, or a valid turtle without a graph, etc).

Finally, a sample of tests cover that the default values also work as before.

To complete, will just try to add tests to `VocabularyConfig`, as I noticed there are methods not being tested there :+1: 

Pull request coming in a few minutes I think.

Cheers
Bruno

**Comment by osma** [comment]
Oh sorry for the mess @kinow, I thought you had replied on GitHub so replied here...

The fix sounds great! Waiting for the PR.

**Comment by kinow** [comment]
My mistake @osma :-) Running the tests one last time, but ready to submit it. Thanks!

**Comment by osma** [comment]
Fixed by PR #793. Also backported to v2.0-maintenance.
